contracts/swap/factory.go

```diff
@@ -0,0 +1,123 @@
+package swap
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	contractFactory "github.com/ethersphere/go-sw3/contracts-v0-1-1/simpleswapfactory"
+)
+
+var (
+	// ErrNotDeployedByFactory is given when a contract was not deployed by the factory
+	ErrNotDeployedByFactory = errors.New("not deployed by factory")
+
+	// Deployments maps from network ids to deployed contract factories
+	Deployments = map[uint64]common.Address{
+		// Ropsten
+		3: common.HexToAddress("0x2e9C43E186eaF4fee10799d67e75f8CFc5BA3a0c"),
+	}
+)
+
+type simpleSwapFactory struct {
+	instance *contractFactory.SimpleSwapFactory
+	address  common.Address
+	backend  Backend
+}
+
+// SimpleSwapFactory interface defines the methods available for a factory contract for SimpleSwap
+type SimpleSwapFactory interface {
+	// DeploySimpleSwap deploys a new SimpleSwap contract from the factory and returns the ready to use Contract abstraction
+	DeploySimpleSwap(auth *bind.TransactOpts, issuer common.Address, defaultHardDepositTimeoutDuration *big.Int) (Contract, error)
+	// VerifyContract verifies that the supplied address was deployed by this factory
+	VerifyContract(address common.Address) error
+	// VerifySelf verifies that this is a valid factory on the network
+	VerifySelf() error
+}
+
+// FactoryAt creates a SimpleSwapFactory instance for the given address and backend
+func FactoryAt(address common.Address, backend Backend) (SimpleSwapFactory, error) {
```

Comment:
- In `contracts/swap` a similar functions is called `instanceAt`. I think I prefer this (`instanceAt`), as the name of the file already tells that it is about a `factory`. 
- What do you think about *always* verifying that a contract is a factory before creating a factory instance? That way, we prevent people from creating factory objects, which are at a later point unuseful. In my opinion, a factory object that does not point to a factory on the blockchain is useless, hence verifySelf should be included in the FactoryAt call as an atomic operation, which then makes in redundant to expose VerifySefl.

Reply:
> In contracts/swap a similar functions is called instanceAt. I think I prefer this (instanceAt), as the name of the file already tells that it is about a factory.
The file name is irrelevant for the caller, so a caller of this function would call `swap.FactoryAt`, which is great I think. `InstanceAt` would conflict and then we need a subpackage which I don't think is needed.

Reply:
The idea here was that `swap.FactoryAt` would be the generic entry point for generating a `SimpleSwapFactory` instance where as `VerifySelf` might be implementation specific (if we have to support multiple versions of the chequebook).